### PR TITLE
executor: add show placement scheduling_state

### LINKED ISSUE #28199: Fetch the placement's scheduling state from PD
The `SHOW PLACEMENT` has a result column named `Scheduling_state` .  Unfortunately, PD does not provide any HTTP API to get the scheduling state for the specified table/partition.

PD have a method `IsRegionReplicated` can make it. It should be easy to export it as a HTTP API.

https://github.com/tikv/pd/blob/fd3fc281e22681e4de48407f0b48a1a5da04d4d3/server/schedule/opt/healthy.go#L60-L68

After it, we can compute the scheduling state for each table/partition.

### PR DESCRIPTION
**Pull request by xhebox**
Signed-off-by: xhe <xw897002528@gmail.com>

### What problem does this PR solve?

Issue Number: close #28199

Problem Summary: Support schedule state query for placement rules, together with some revisions on the RFC.

### Check List

Tests 

- [x] Unit test
- [x] Integration test
- [x] Manual test (add detailed scripts or steps below)

```sh
mycli -h 127.0.0.1 -P 4000 -u root --no-warn << "EOF"
use test;
drop table if exists test.table1;
drop placement policy if exists x;
create placement policy x FOLLOWERS=1;
create table test.table1 (id int primary key) placement policy = "x";
insert into test.table1 values (1);
insert into test.table1 values (2);
insert into test.table1 values (3);
insert into test.table1 values (4);
insert into test.table1 values (5);
show placement;
show placement for database test;
show placement for table table1;
EOF
```

### Release note

```release-note
Support query scheduling state using `show placement ...` statements.
```

### CODE REVIEW

File: docs/design/2020-06-24-placement-rules-in-sql.md [hunk 1]
```diff
@@ -229,8 +229,8 @@ SHOW PLACEMENT;
 +----------------------------+----------------------------------------------------------------------+------------------+
 | target                     | placement                                                            | scheduling_state |
 +----------------------------+----------------------------------------------------------------------+------------------+
-| POLICY system              | PRIMARY_REGION="us-east-1" REGIONS="us-east-1,us-east-2" FOLLOWERS=4 | SCHEDULED        |
-| POLICY default             | PRIMARY_REGION="us-east-1" REGIONS="us-east-1,us-east-2"             | SCHEDULED        |
+| POLICY system              | PRIMARY_REGION="us-east-1" REGIONS="us-east-1,us-east-2" FOLLOWERS=4 |                  |
```
Comment (lcwangchao) [inline docs/design/2020-06-24-placement-rules-in-sql.md:232]: @morgo What do you think about the scheduling_state for a policy. Some options: `SCHEDULED ` , `NULL` or an empty string just like this pr did, which is better? Actually we do not schedule policy in PD.

Comment (morgo) [reply docs/design/2020-06-24-placement-rules-in-sql.md:232]: Null is ideal (since it means not applicable)

File: executor/show_placement.go [hunk 2]
```diff
@@ -387,3 +418,68 @@ func (e *ShowExec) getPolicyPlacement(policyRef *model.PolicyRefInfo) (settings
...
 	return policy.PlacementSettings, nil
 }
+
+func fetchScheduled(ctx context.Context, scheduleState map[int64]bool, id int64) (bool, error) {
+	if s, ok := scheduleState[id]; ok {
+		return s, nil
+	}
+	startKey := codec.EncodeBytes(nil, tablecodec.GenTablePrefix(id))
+	endKey := codec.EncodeBytes(nil, tablecodec.GenTablePrefix(id+1))
+	schedule, err := infosync.GetReplicationState(ctx, startKey, endKey)
```
Comment (AilinKid) [inline executor/show_placement.go:428]: is there a batch interface?  one request for a single seems costy

**Review by AilinKid — APPROVED** [review]
LGTM

File: executor/show_placement.go (hunk 2 above)
Comment (AilinKid) [reply executor/show_placement.go:428]: given the emergency, let merge it first

Comment (xhebox) [reply executor/show_placement.go:428]: No, there is no batch api.

**Pull request merged**